scaped parameter cause doc warnings

```
/// <param name="@params">The endpoint parameters.</param>
```
Produces this warning

```
D:\IceRPC\icerpc-csharp\src\IceRpc\generated\IceRpc\Endpoint.cs(112,26): warning CS1572: XML comment has a param tag for '@params', but there is no parameter by that name [D:\IceRPC\icerpc-csharp\src\IceRpc\IceRpc.csproj]
         D:\IceRPC\icerpc-csharp\src\IceRpc\generated\IceRpc\Endpoint.cs(113,148): warning CS1573: Parameter 'params' has no matching param tag in the XML comment for 'EndpointData.EndpointData(Protocol, string, string, ushort, IList<EndpointParam>)' (but other parameters do) [D:\IceRPC\icerpc-csharp\src\IceRpc\IceRpc.csproj]
```

The name must not be escaped in the XML comment

```
/// <summary>The endpoint parameters.</summary>
public readonly global::System.Collections.Generic.IList<EndpointParam> @Params;
```
This is also not required as the keywords are case sensitive and `Params` is not reserved.

--- a/cpp/src/slice2cs/CsUtil.cpp
+++ b/cpp/src/slice2cs/CsUtil.cpp
@@ -38,10 +38,18 @@ Slice::operationName(const OperationPtr& op)
 }
 
 std::string
-Slice::paramName(const MemberPtr& param, const string& prefix)
+Slice::paramName(const MemberPtr& param, const string& prefix, bool escapeKeywords)
 {
     string name = param->name();
-    return normalizeCase(param) ? fixId(prefix + camelCase(name)) : fixId(prefix + name);
+
+    if (normalizeCase(param))
+    {
+        name = camelCase(name);
+    }
+
+    name = prefix + name;
+
+    return escapeKeywords ? fixId(name) : name;
 }
 
 std::string

--- a/cpp/src/slice2cs/CsUtil.h
+++ b/cpp/src/slice2cs/CsUtil.h
@@ -31,7 +31,7 @@ std::string getUnqualified(const ContainedPtr&,
 
 bool normalizeCase(const ContainedPtr&);
 std::string operationName(const OperationPtr&);
-std::string paramName(const MemberPtr& param, const std::string& prefix = "");
+std::string paramName(const MemberPtr& param, const std::string& prefix = "", bool escapeKeywords = true);
 std::string paramTypeStr(const MemberPtr& param, const std::string& ns, bool readOnly = false);
 
 std::string fieldName(const MemberPtr&);

--- a/cpp/src/slice2cs/Gen.cpp
+++ b/cpp/src/slice2cs/Gen.cpp
@@ -972,7 +972,7 @@ Slice::CsVisitor::writeParamDocComment(const OperationPtr& op, const CommentInfo
         auto i = comment.params.find(param->name());
         if(i != comment.params.end())
         {
-            writeDocCommentLines(_out, i->second, "param", "name", fixId(param->name()));
+            writeDocCommentLines(_out, i->second, "param", "name", paramName(param, "", false));
         }
     }
 }
@@ -1223,7 +1223,7 @@ Slice::Gen::TypesVisitor::visitClassDefEnd(const ClassDefPtr& p)
         for (const auto& member : allDataMembers)
         {
             CommentInfo comment = processComment(member, "");
-            writeDocCommentLines(_out, comment.summaryLines, "param", "name", paramName(member));
+            writeDocCommentLines(_out, comment.summaryLines, "param", "name", paramName(member, "", false));
         }
         _out << nl << "public " << name
              << spar
@@ -1271,7 +1271,7 @@ Slice::Gen::TypesVisitor::visitClassDefEnd(const ClassDefPtr& p)
             for (const auto& member : allMandatoryDataMembers)
             {
                 CommentInfo comment = processComment(member, "");
-                writeDocCommentLines(_out, comment.summaryLines, "param", "name", paramName(member));
+                writeDocCommentLines(_out, comment.summaryLines, "param", "name", paramName(member, "", false));
             }
             _out << nl << "public " << name
                 << spar
@@ -1520,7 +1520,7 @@ Slice::Gen::TypesVisitor::visitExceptionEnd(const ExceptionPtr& p)
             for (const auto& member : p->allDataMembers())
             {
                 CommentInfo comment = processComment(member, "");
-                writeDocCommentLines(_out, comment.summaryLines, "param", "name", paramName(member));
+                writeDocCommentLines(_out, comment.summaryLines, "param", "name", paramName(member, "", false));
             }
 
             if (i > 0)
@@ -1684,7 +1684,7 @@ Slice::Gen::TypesVisitor::visitStructEnd(const StructPtr& p)
     for (const auto& member : dataMembers)
     {
         CommentInfo comment = processComment(member, "");
-        writeDocCommentLines(_out, comment.summaryLines, "param", "name", paramName(member));
+        writeDocCommentLines(_out, comment.summaryLines, "param", "name", paramName(member, "", false));
     }
     _out << nl << "public ";
     _out << name

